core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -98,20 +100,57 @@ public BlockLockManager() {}
    * @return lock id
    */
   public long lockBlock(long sessionId, long blockId, BlockLockType blockLockType) {
+    return lockBlockInternal(sessionId, blockId, blockLockType, true, null, null);
+  }
+
+  /**
+   * Tries to lock a block within the given time.
+   * Note that even if this block does not exist, a lock id is still returned.
+   *
+   * If all {@link PropertyKey#WORKER_TIERED_STORE_BLOCK_LOCKS} are already in use and no lock has
+   * been allocated for the specified block, this method will need to wait until a lock can be
+   * acquired from the lock pool.
```

Comment:
This will either wait for a free lock, or timeout (if the timeout is set), right?